src/components/calcite-dropdown-item/calcite-dropdown-item.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Prop,
+  State
+} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+import DropdownInterface from "../../interfaces/DropdownInterface";
+
+@Component({
+  tag: "calcite-dropdown-item",
+  styleUrl: "calcite-dropdown-item.scss",
+  shadow: true
+})
+export class CalciteDropdownItem {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  @Prop({ reflect: true, mutable: true }) active: boolean = false;
+
+  /** @internal */
+  @Prop() currentDropdownGroup: string = this.el.parentElement.id;
+
+  /** @internal */
+  @Prop() requestedDropdownGroup: string = "";
+
+  /** @internal */
+  @Prop() requestedDropdownItem: string = "";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteDropdownItemSelected: EventEmitter;
+  @Event() calciteDropdownItemKeyEvent: EventEmitter;
+  @Event() closeCalciteDropdown: EventEmitter;
+  @Event() registerCalciteDropdownItem: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentDidLoad() {
+    this.currentDropdownGroup = this.el.parentElement.id;
+    this.itemPosition = this.getItemPosition();
+
+    this.registerCalciteDropdownItem.emit({
+      item: this.el,
+      position: this.itemPosition
+    });
+  }
+
+  componentDidUpdate() {
+    this.determineActiveItem();
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    const selected = this.active ? "true" : null;
+    return (
+      <Host
+        dir={dir}
+        id={this.dropdownItemId}
+        tabindex="0"
+        role="menuitem"
+        aria-selected={selected}
+      >
+        <slot />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("click") onClick(e) {
+    this.emitRequestedItem(e);
+  }
+
+  @Listen("keydown") keyDownHandler(e) {
+    switch (e.key) {
```

Comment:
I think unfortunately `event.key` is not totally supported by IE11, so `keyCode` is safer. Are we targeting that browser?